src/colab/Run_SKAI_Colab_Pipeline.ipynb

```diff
@@ -0,0 +1,749 @@
+        "\n",
+        "#@markdown ---\n",
+        "SKAI_CODE_DIR = \"/content/skai_src\"  #@param {type:\"string\"}\n",
+        "SKAI_VENV_DIR = \"/content/skai_env\"  #@param {type:\"string\"}\n",
+        "SKAI_REPO = \"https://github.com/google-research/skai.git\"  #@param {type:\"string\"}\n",
+        "SKAI_BRANCH = \"main\"  #@param {type:\"string\"}\n",
+        "SKAI_COMMIT = \"\" #@param {type:\"string\"}\n",
+        "\n",
+        "root_filesys=os.path.dirname(SKAI_CODE_DIR)\n",
+        "\n",
+        "pathsys_venv=SKAI_VENV_DIR\n",
+        "pathsys_actenv=os.path.join(pathsys_venv, 'bin/activate')\n",
+        "\n",
+        "pathsys_skai=SKAI_CODE_DIR\n",
+        "%shell rm -rf {SKAI_CODE_DIR}\n",
+        "%shell git clone -b {SKAI_BRANCH} {SKAI_REPO} {SKAI_CODE_DIR}\n",
+        "if SKAI_COMMIT!='':\n",
+        "  %shell cd {SKAI_CODE_DIR} ; git checkout {SKAI_COMMIT}\n",
+        "\n",
+        "%cd {SKAI_CODE_DIR}/src/colab\n",
+        "import colab_utils\n",
+        "from colab_utils import *\n",
+        "\n",
+        "def launch_pexpect_process(script, arguments,dir_args, use_pexpect, sleep=None):\n",
+        "  if not isinstance(script, list):\n",
+        "    script=[script]\n",
+        "    arguments=[arguments]\n",
+        "  \n",
+        "  if sleep==None and len(script)>1:\n",
+        "    sleep=[0] * (len(script)-1)\n",
+        "  elif len(script)==1:\n",
+        "    sleep==None\n",
+        "\n",
+        "  flags_str = [' '.join(f\"--{f}='{v}'\" for f, v in argument.items()) for argument in arguments] \n",
+        "  commands = '; '.join([\n",
+        "      f'set -e',\n",
+        "      f'source {dir_args[\"python_env\"]}',\n",
+        "      f'export GOOGLE_APPLICATION_CREDENTIALS={dir_args[\"path_cred\"]}',\n",
+        "      f'python {dir_args[\"path_skai\"]}/src/{script[0]} {flags_str[0]}'])\n",
+        "  \n",
+        "  if sleep is not None:\n",
+        "    commands_bis=' '.join([\n",
+        "      '; '.join([f'& sleep {sleep[i-1]} ',f'python {dir_args[\"path_skai\"]}/src/{script[i]} {flags_str[i]}']) for i in range(1,len(script))])\n",
+        "    \n",
+        "    commands=' '.join([commands, commands_bis])\n",
+        "\n",
+        "  \n",
+        "  sh_command = f'bash -c \"{commands}\" | tee /tmp/output.txt'\n",
+        "  print(sh_command,'\\n')\n",
+        "\n",
+        "  if use_pexpect:\n",
+        "    return pexpect.spawn(sh_command)\n",
+        "  else:\n",
+        "    with open('/tmp/shell_command.sh', 'w') as f:\n",
+        "      f.write(commands)\n",
+        "    !bash \"/tmp/shell_command.sh\" | tee /tmp/output.txt\n",
+        "\n",
+        "def load_start_tensorboard(path_log):\n",
+        "  %load_ext tensorboard\n",
+        "  %tensorboard --logdir gs://{path_log}\n",
+        "\n",
+        "colab_utils.launch_pexpect_process=launch_pexpect_process\n",
```

Comment:
This is a bit too tricky and error prone. Please make the change I recommended above and then move "launch_pexpect_process" into colab_utils.py.

Not sure how to fix "load_start_tensorboard". You can leave that for now and we can fix it later.